### PR TITLE
Adding demo with basic features

### PR DESCRIPTION
**Pull request by danielaRiesgo**
Adding visual demo with:
- NibLoadable and loadChildController(into:)
- ViewGradient

Issue #86

### CODE REVIEW

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 86.69
:x: Code Quality (Swiftlint): 82.0 - Minimum: 90.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/add-visual-demo) for more details.

**Comment by pablog** [comment]
@danielaRiesgo you can ignore the demo project in `slather.yml` if you want.

**Comment by pablog** [comment]
I'd also remove the demo and demoUI targets. We just want to test the framework.

File: CoreDemo/ChildView.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//  CoreDemo
+//
+//  Created by Daniela Riesgo on 12/19/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import UIKit
+import Core
+
+class ChildView: UIView, NibLoadable {
```
Comment (pablog) [inline CoreDemo/ChildView.swift]: `final internal`

File: CoreDemo/ChildController.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+//  CoreDemo
+//
+//  Created by Daniela Riesgo on 12/19/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import UIKit
+import Core
+
+class ChildController: UIViewController {
```
Comment (pablog) [inline CoreDemo/ChildController.swift]: `final internal`

File: CoreDemo/ChildController.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    lazy var _view: ChildView = ChildView.loadFromNib()!
+
+    override func loadView() {
+        view = _view
+    }
+
+    override func viewDidLoad() {
```
Comment (pablog) [inline CoreDemo/ChildController.swift]: Remove this function.

File: CoreDemo/ChildController.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+    }
+
+    override func didReceiveMemoryWarning() {
```
Comment (pablog) [inline CoreDemo/ChildController.swift]: Remove this function.

File: CoreDemo/View.swift [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+//  CoreDemo
+//
+//  Created by Daniela Riesgo on 12/19/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import UIKit
+import Core
+
+class View: UIView, NibLoadable {
```
Comment (pablog) [inline CoreDemo/View.swift]: `final internal`

File: CoreDemo/ViewController.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+//  CoreDemo
+//
+//  Created by Daniela Riesgo on 12/19/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import UIKit
+import Core
+
+class ViewController: UIViewController {
```
Comment (pablog) [inline CoreDemo/ViewController.swift]: `final internal`

File: CoreDemo/ViewController.swift [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        view = _view
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+        load(childViewController: _childController, into: _view.childContainerView)
+    }
+
+    override func didReceiveMemoryWarning() {
```
Comment (pablog) [inline CoreDemo/ViewController.swift]: Remove this function.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 86.69
:white_check_mark: Code Quality (Swiftlint): 92.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/add-visual-demo) for more details.

**Pull request merged**